plugins/hls-tactics-plugin/src/Ide/Plugin/Tactic/LanguageServer/TacticProviders.hs

```diff
@@ -54,6 +54,40 @@ commandTactic UseDataCon             = userSplit
 commandTactic Refine                 = const refine
 
 
+------------------------------------------------------------------------------
+-- | The LSP kind
+tacticKind :: TacticCommand -> T.Text
+tacticKind Auto                   = "fillHole"
+tacticKind Intros                 = "introduceLambda"
+tacticKind Destruct               = "caseSplit"
+tacticKind Homomorphism           = "homomorphicCaseSplit"
+tacticKind DestructLambdaCase     = "lambdaCase"
+tacticKind HomomorphismLambdaCase = "homomorphicLambdaCase"
+tacticKind DestructAll            = "splitFuncArgs"
+tacticKind UseDataCon             = "useConstructor"
+tacticKind Refine                 = "refine"
+
+
+------------------------------------------------------------------------------
+-- | Whether or not this code action is preferred -- ostensibly refers to
+-- whether or not we can bind it to a key in vs code?
+tacticPreferred :: TacticCommand -> Bool
+tacticPreferred Auto                   = True
+tacticPreferred Intros                 = True
+tacticPreferred Destruct               = True
+tacticPreferred Homomorphism           = False
+tacticPreferred DestructLambdaCase     = False
+tacticPreferred HomomorphismLambdaCase = False
+tacticPreferred DestructAll            = True
+tacticPreferred UseDataCon             = True
+tacticPreferred Refine                 = True
+
+
+mkTacticKind :: TacticCommand -> CodeActionKind
+mkTacticKind =
+  CodeActionUnknown . mappend "refactor.wingman." . tacticKind
```

Comment:
I'm not familiar to these things, but since tactics plugin resolves (or changes) some errors, it may be better to use `quickfix.wingman.`. WDYT?

Reply:
They don't seem very quickfix-y to me, and I think a bunch of them work with holes, so they won't actually be fixing errors.

Reply:
@michaelpj I mean a hole produces a compile error, so... But I'm not strongly inclined to anything :)